sphinx/ext/autodoc/__init__.py

```diff
@@ -530,6 +530,9 @@ def add_directive_header(self, sig: str) -> None:
             # Be explicit about the module, this is necessary since .. class::
             # etc. don't support a prepended module name
             self.add_line('   :module: %s' % self.modname, sourcename)
+        if self.options.canonical or self.fullname != self.canonical_fullname:
```

Comment:
`:canonical:` option is not defined in auto* directives. So we can't use `self.options.canonical` here.

Reply:
Artifact of a local exploration, sorry about that